### DISCUSSION TITLE
CoNLL-U Format or ?

### DISCUSSION THREAD

**Discussion by yosiasz** [explosion/spaCy#6648] [Help: Coding & Implementations] [answered]
Greetings,

1. Working on developing extensive annotations for both Amharic and Tigrinya. Is  CoNLL-U the only format for annotations or are there other formats I can use?
2. Can I use https://github.com/UniversalDependencies/UD_Amharic-ATT to create Amharic spacy-models?

Thanks

**Comment by adrianeboyd** [comment, marked as answer]
You can use any format that you can convert to a spacy `Doc`, but it's easier if you pick a format that is supported by annotation tools and has a spacy converter, so CoNLL-U is a good choice.

The UD_Amharic-ATT corpus could be a good starting point, but it's probably too small on its own to train models with useful performance.

The large number of multi-word tokens could also be an issue, since a `Doc` only supports one layer of tokenization. If you do try it out, you'd probably want to convert the corpus using `spacy convert --merge-subtokens` so that the tokens in the training data correspond to the merged rather than the split forms, which is what you'd get as input as a plain text in your spacy pipeline. You'd have to see whether the distinctions that are lost in the merged forms cause problems for downstream tasks. If you do decide to use this particular UD annotation scheme and want to keep these distinctions, you might want to look at a tool like udpipe or stanza instead of spacy, which can handle the UD multi-word annotation better.

Before you start training, you'd also want to confirm that spacy's tokenizer output is very similar to the tokenization in the corpus. You probably want `token_acc` to be at least 95% or so before you start training because the models can't learn well when there a lot of misaligned tokens. (For the provided pipelines, I've tried to get `token_acc` to >99.5%.)
